api/http/handler/stacks/stack_migrate.go

```diff
@@ -89,11 +90,17 @@ func (handler *Handler) stackMigrate(w http.ResponseWriter, r *http.Request) *ht
 		stack.SwarmID = payload.SwarmID
 	}
 
+	oldName := stack.Name
```

Comment:
Why do you keep a reference to the previous stack name here? And why do you re-assign it after? 

There are only 2 cases here: 

* we specify a name for the migration and it should be used (`payload.Name != ""`)
* we do not specify a name for the migration and it keeps the old name (no need to reference it)

Reply:
because if I change `stack.name` then on delete it tries to delete a stack with the new name. Might be better to pass to deletestack only the relevant parameters (stack name and anything else it needs), but I think this function is used in different places so it expects the stack object

Reply:
Makes sense.